Desciption:  Local Privilege Escalation via host option
 Sudo's host (`-h` or `--host`) option is intended to be used in
 conjunction with the list option (`-l` or `--list`) to list a user's
 sudo privileges on a host other than the current one.  However, due
 to a bug it was not restricted to listing privileges and could be
 used when running a command via `sudo` or editing a file with
 `sudoedit`.  Depending on the rules present in the sudoers file
 this could allow a local privilege escalation attack.
Forwarded: not-needed

--- a/plugins/sudoers/sudoers.c
+++ b/plugins/sudoers/sudoers.c
@@ -467,6 +467,18 @@ sudoers_policy_main(int argc, char * con
 	sudoers_gc_add(GC_PTR, NewArgv[0]);
     }
 
+    /* The user may only specify a host for "sudo -l". */
+    if (!ISSET(sudo_mode, MODE_LIST|MODE_CHECK)) {
+	if (strcmp(user_runhost, user_host) != 0) {
+	    log_warningx(SLOG_NO_STDERR|SLOG_AUDIT,
+		N_("user not allowed to set remote host for command"));
+	    sudo_warnx("%s",
+		U_("a remote host may only be specified when listing privileges."));
+	    ret = false;
+	    goto done;
+	}
+    }
+
     /* If given the -P option, set the "preserve_groups" flag. */
     if (ISSET(sudo_mode, MODE_PRESERVE_GROUPS))
 	def_preserve_groups = true;
